### DISCUSSION TITLE
Tailwind CSS: Cannot read properties of undefined (reading 'modifier')

### DISCUSSION THREAD

**Discussion by tlerbao** [tailwindlabs/tailwindcss#9634] [Help]
Project: vue3+vite+ts, created by npm init vue@latest
Software: phpstorm
Problems and errors:
The style has taken effect, but the ide does not have any intellisense and reports an error
Error notification:
<img width="525" alt="image" src="https://user-images.githubusercontent.com/20442232/197330540-4ea90528-b10b-44c2-97e6-a558df1ac5c8.png">
<img width="406" alt="image" src="https://user-images.githubusercontent.com/20442232/197330564-8d27bf43-a4f7-4de6-be50-be1576a68ce4.png">

**Comment by wongjn** [comment]
It could be that the built-in Tailwind plugin for PHPStorm has not been updated for Tailwind v3.2, if you have that version installed.

Edit: [Here's the related ticket on YouTrack](https://youtrack.jetbrains.com/issue/WEB-57745/Tailwind-no-completion-when-using-tailwindcss320-Cannot-read-properties-of-undefined-reading-modifier). Relevant comments from JetBrains staff:
> As of now, WebStorm and IntelliJ IDEA 2022.3 EAPs with the fix are available. At the moment I don't know the exact date of the next PhpStorm EAP. IDE build number with the fix will be 223.7255.* or later.  
> The plan is also to backport the fix to the next 2022.2.x update release. Its release date is not yet known.[^1]

> Yes, in addition to version 2022.3 (where the fix is already available in public EAP builds), this will also be fixed in WebStorm (and other IDEs) version 2022.2.4 (expected later this month).[^2]

[^1]: https://youtrack.jetbrains.com/issue/WEB-57745/Tailwind-no-completion-when-using-tailwindcss320-Cannot-read-properties-of-undefined-reading-modifier#focus=Comments-27-6566205.0-0
[^2]: https://youtrack.jetbrains.com/issue/WEB-57745/Tailwind-no-completion-when-using-tailwindcss320-Cannot-read-properties-of-undefined-reading-modifier#focus=Comments-27-6574422.0-0

**Reply by tlerbao** [reply]
@wongjn Thanks, sure enough, installing 3.1 is no problem;
Take another leave, can you ignore the following question?
I am a newbie.

<img width="608" alt="image" src="https://user-images.githubusercontent.com/20442232/197339113-2406d733-487d-45cc-909b-4e5b8bbeabd6.png">

**Reply by kusime** [reply]
this error can be fixed by edit the .eslintrc.cjs file ,add this config below then watch out !

  "env": {
    "node": true,
    "browser": true,
    "commonjs": true,
    "amd": true
  },

**Reply by tlerbao** [reply]
@kusime thanks

**Comment by CTOverton** [comment]
Found this post because in Webstorm tailwind auto completes weren't working, downgraded Tailwind to `tailwindcss@3.0.5` and autocompletes are back.

Assuming @wongjn is right and that Jetbrains just hasn't updated the tailwind plugin to support Tailwind 3.2

**Reply by tlerbao** [reply]
@CTOverton Yes, I use 3.1.8 without error

**Reply by CTOverton** [reply]
Hopefully they update it soon, always annoying when it just stops working and seemly with no explanation :P

**Reply by Xevion** [reply]
So frustrating! Seems like while developing my project the plugin stops working for no apparent reason. And since I'm not using Tailwind autocomplete constantly, it might take effect without me noticing, leading to a bad diagnosis. Can confirm, `3.1.8` is a valid working version on Webstorm 2022.2 (latest as of 11/5/2022).

**Reply by wongjn** [reply]
[Here's the related ticket on YouTrack](https://youtrack.jetbrains.com/issue/WEB-57745/Tailwind-no-completion-when-using-tailwindcss320-Cannot-read-properties-of-undefined-reading-modifier). Relevant comments from JetBrains staff:
> As of now, WebStorm and IntelliJ IDEA 2022.3 EAPs with the fix are available. At the moment I don't know the exact date of the next PhpStorm EAP. IDE build number with the fix will be 223.7255.* or later.  
> The plan is also to backport the fix to the next 2022.2.x update release. Its release date is not yet known.[^1]

> Yes, in addition to version 2022.3 (where the fix is already available in public EAP builds), this will also be fixed in WebStorm (and other IDEs) version 2022.2.4 (expected later this month).[^2]

[^1]: https://youtrack.jetbrains.com/issue/WEB-57745/Tailwind-no-completion-when-using-tailwindcss320-Cannot-read-properties-of-undefined-reading-modifier#focus=Comments-27-6566205.0-0
[^2]: https://youtrack.jetbrains.com/issue/WEB-57745/Tailwind-no-completion-when-using-tailwindcss320-Cannot-read-properties-of-undefined-reading-modifier#focus=Comments-27-6574422.0-0
